### PR TITLE
746 post processing should follow row_split_var in main function

### LINKED ISSUE #746: post processing should follow row_split_var in main function
https://github.com/insightsengineering/chevron/blob/ea38d6c137045683179e81ea2d9ca5aac74cd4d2/R/mht01.R#L108

if row_split_var is not MHBODSYS, then the post-processing will fail

### PR DESCRIPTION
**Pull request by BFalquet**
https://github.com/insightsengineering/chevron/issues/746

close #746 

thank you for the review

### CODE REVIEW

File: R/mht01.R [hunk 1]
```diff
@@ -105,14 +105,14 @@ mht01_pre <- function(adam_db, ...) {
...
-mht01_post <- function(tlg, prune_0 = TRUE, ...) {
+mht01_post <- function(tlg, row_split_var = "MHBODSYS", prune_0 = TRUE, ...) {
   if (prune_0) {
     tlg <- smart_prune(tlg)
   }
 
   tbl_sorted <- tlg %>%
     sort_at_path(
-      path = c("MHBODSYS", "*", "MHDECOD"),
+      path = c(row_split_var, "*", "MHDECOD"),
```
Comment (clarkliming) [inline R/mht01.R:115]: if there are multiple `row_split_var`, will this still work?

Comment (BFalquet) [reply R/mht01.R]: good point, now it does.

File: R/mht01.R [hunk 2]
```diff
@@ -105,14 +105,16 @@ mht01_pre <- function(adam_db, ...) {
...
   if (prune_0) {
     tlg <- smart_prune(tlg)
   }
 
+  row_split_var <- c(rbind(row_split_var, "*"))
+
   tbl_sorted <- tlg %>%
     sort_at_path(
-      path = c("MHBODSYS", "*", "MHDECOD"),
+      path = c(row_split_var, "MHDECOD"),
```
Comment (clarkliming) [inline R/mht01.R:117]: should be split1, *, split 2, *, split 3, * mhdecod?

Comment (BFalquet) [reply R/mht01.R:117]: I added the `row_split_var <- c(rbind(row_split_var, "*"))` just above to do that. It is a bit funny but apparently the shortest way to alternate elements from two different vectors.

**Review by clarkliming — APPROVED** [review]
looks good! thank you

**Pull request merged**